src/tools.ts

```diff
@@ -17,15 +17,17 @@ export default class Tools {
 
             //Check for files >100mb on GitHub remote
             if (remoteUrl?.includes("github.com")) {
-                const tooBigFiles = files.filter((f) => {
-                    const file = this.plugin.app.vault.getAbstractFileByPath(
-                        f.vault_path
-                    );
+                const tooBigFiles = [];
+                
+                for (const f of files) {
+                    const file = this.plugin.app.vault.getAbstractFileByPath(f.vault_path);
                     if (file instanceof TFile) {
-                        return file.stat.size >= 100000000;
+                        if (file.stat.size >= 100000000 && !(await this.plugin.gitManager.isFileTrackedByLFS(f.vault_path))) {
```

Comment:
By only implementing the method in `SimpleGit`  you  will need a check whether the `gitManager` is an instance of that class. And please try to make this line shorter, e.g. by extracting this to an extra variable.